demo-app/src/main/java/org/openmined/syft/demo/ui/FederatedCycleViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package org.openmined.syft.demo.ui
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import org.openmined.syft.Syft
+import org.openmined.syft.networking.clients.HttpClient
+import org.openmined.syft.networking.clients.SocketClient
+import org.openmined.syft.networking.datamodels.ClientConfig
+import org.openmined.syft.processes.JobStatusSubscriber
+import org.openmined.syft.threading.ProcessSchedulers
+
+private const val TAG = "FederatedCycleViewModel"
+
+@ExperimentalUnsignedTypes
+class FederatedCycleViewModel(
+    socketClient: SocketClient,
```

Comment:
I would recommend to use the repository pattern here. We can leave it if you feel it's going to be too much work now but let's open an issue in that case.
I'll leave the decision to you.

Reply:
Thanks for the info. I will create an issue for that. This code needs to be used for other issues as well.